### PR TITLE
Feature/cri 685 batch fetcher operation

### PR DESCRIPTION
**Pull request by mmiedlarz**

### CODE REVIEW

File: CreatubblesAPIClient/Sources/Network/OAuth2Client.swift [hunk 1]
```diff
@@ -25,31 +25,35 @@
...
+//        get {
+//            return keychain["public-access-token"]
+//        }
+//        
+//        set {
+//            keychain["public-access-token"] = newValue
+//        }
+//    }
+//    
     func logout() {
```
Comment (mmiedlarz) [inline CreatubblesAPIClient/Sources/Network/OAuth2Client.swift:53]: remember to not commit this file!

File: CreatubblesAPIClient/Sources/DAO/BatchFetch/Creations/CreationsBatchFetchOperation.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        self.partnerApplicationId = partnerApplicationId
+        self.sort = sort
+        self.onlyPublic = onlyPublic
+        self.pagingData = pagingData
+        
+        super.init(complete: complete)
+    }
+    
+    override func main()
+    {
```
Comment (pleszkiewicz) [inline CreatubblesAPIClient/Sources/DAO/BatchFetch/Creations/CreationsBatchFetchOperation.swift:57]: I'd check if the operation is already cancelled as often as possible. The beginning of the main function is a good place for it.

File: CreatubblesAPIClient/Sources/DAO/BatchFetch/Creations/CreationsBatchFetchOperation.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        let request =  FetchCreationsRequest(page: pagingData.page, perPage: pagingData.pageSize, galleryId: galleryId, userId: userId, sort: sort, keyword: keyword, partnerApplicationId: partnerApplicationId, onlyPublic: onlyPublic)
+        let handler = FetchCreationsResponseHandler()
+        {
+            [weak self](creations, pagingInfo, error) -> (Void) in
+            self?.creations = creations
+            self?.finish(error)
+        }
+        requestSender.send(request, withResponseHandler: handler)
+    }
```
Comment (pleszkiewicz) [inline CreatubblesAPIClient/Sources/DAO/BatchFetch/Creations/CreationsBatchFetchOperation.swift:74]: Would be good to override `cancel` method to stop current request from being performed.
```
override func cancel()
{
    self.request?.cancel() // not sure what should be cancelled here: request, requestSender or handler?
    super.cancel()
}
```

**Review by pleszkiewicz — CHANGES_REQUESTED** [review]
Code: +1 (`cancel` method should be overridden)
Func: n/a

**Pull request closed without merging**